File: level2/주식가격.py
def solution(prices):
    size = len(prices)
    answer = [0]*size
    
    s = []
    for i in range(size):
        while s:
            if prices[i] >= prices[s[-1]]:
                break
            answer[s[-1]] = i-s[-1]
            s.pop()
        s.append(i)
        
    while s:
        answer[s[-1]] = size-s[-1]-1
        s.pop()
    
    return answer

File: C++/주식가격.cpp
#include <stack>
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> prices) {
    int size = prices.size();
    
    vector<int> answer(size);
    stack<int> s;
    for(int i = 0; i < size; i++){
        while(!s.empty()){
            if(prices[s.top()] <= prices[i]){
                break;
            }
            answer[s.top()] = i-s.top();
            s.pop();
        }
        s.push(i);
    }
    
    while(!s.empty()){
        answer[s.top()] = size-s.top()-1;
        s.pop();
    }
    
    return answer;
}
